### PR TITLE
Chef can now name meals

### PR DESCRIPTION
**Pull request by Valtsu0**
## About the PR 
Allows chefs (and anyone with chef training) to name meals by using a plate in hand. People without chef training will still have the old in hand use effect of dumping a random piece of food.
image
image
image

## Why's this needed? 
Implements [this thread](https://forum.ss13.co/showthread.php?tid=21540)

## Changelog 

```changelog
(u)Valtsu0
(+)Chef can now name meals by using a plate in hand
```

### CODE REVIEW

File: code/obj/item/kitchen.dm [hunk 1]
```diff
@@ -766,10 +766,29 @@ TRAYS
...
+			tooltip_rebuild = 1
+			var/holder = src.loc
+			var/str = copytext(html_encode(tgui_input_text(user, "Dish name?", "Set name")), 1, 32)
+			if(str)
+				phrase_log.log_phrase("dish_name", str, no_duplicates=TRUE)
+			if (src.loc != holder)
+				return
+			if(url_regex?.Find(str))
+				return
+			if (length(str) > 64)
```
Comment (TobleroneSwordfish) [inline code/obj/item/kitchen.dm:783]: you're cutting the string to length 32, then checking if it's longer than 64?

File: code/obj/item/kitchen.dm [hunk 2]
```diff
@@ -766,10 +766,29 @@ TRAYS
...
+	// chefs are too fancy for that and will instead get to name the dish
+	attack_self(mob/user)
 		. = ..()
-		if (length(src.contents))
-			src.remove_contents(pick(src.contents))
+		if(user.traitHolder?.hasTrait("training_chef"))
+			tooltip_rebuild = 1
+			var/holder = src.loc
+			var/str = copytext(html_encode(tgui_input_text(user, "Dish name?", "Set name")), 1, 32)
+			if(str)
```
Comment (TobleroneSwordfish) [inline code/obj/item/kitchen.dm:777]: better to just return if `!length(str)`

File: code/obj/item/kitchen.dm [hunk 3]
```diff
@@ -766,10 +766,29 @@ TRAYS
...
+	// in case you only have one arm or you stacked too many MONSTERs or something just dump a random piece of food
+	// chefs are too fancy for that and will instead get to name the dish
+	attack_self(mob/user)
 		. = ..()
-		if (length(src.contents))
-			src.remove_contents(pick(src.contents))
+		if(user.traitHolder?.hasTrait("training_chef"))
+			tooltip_rebuild = 1
+			var/holder = src.loc
+			var/str = copytext(html_encode(tgui_input_text(user, "Dish name?", "Set name")), 1, 32)
```
Comment (TobleroneSwordfish) [inline code/obj/item/kitchen.dm:776]: `tgui_input_text` takes a `max_length` argument for preventing the user from entering text past a certain length (you should still check again on the server though)

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: code/obj/item/kitchen.dm [hunk 4]
```diff
@@ -1206,4 +1229,4 @@ TRAYS
 	icon = 'icons/obj/kitchen.dmi'
 	icon_state = "tongs"
 
-	// used in attackby procs of /obj/item/reagent_containers/food/snacks and /obj/item/kitchen/food_box
+	// used in attackby procs ofgent_coners/food/snacks and /obj/item/kitchen/food_box
```
Comment (frawhst) [inline code/obj/item/kitchen.dm:1232]: I dunno what happened here but it probably shouldn't have

File: code/obj/item/kitchen.dm [hunk 5]
```diff
@@ -766,10 +766,33 @@ TRAYS
...
 
-	attack_self(mob/user) // in case you only have one arm or you stacked too many MONSTERs or something just dump a random piece of food
+	// in case you only have one arm or you stacked too many MONSTERs or something just dump a random piece of food
+	// chefs are too fancy for that and will instead get to name the dish
+	attack_self(mob/user)
 		. = ..()
-		if (length(src.contents))
-			src.remove_contents(pick(src.contents))
+		if(user.traitHolder?.hasTrait("training_chef"))
+			tooltip_rebuild = 1
```
Comment (frawhst) [inline code/obj/item/kitchen.dm:774]:
```suggestion
			tooltip_rebuild = TRUE
```
tiny change, `TRUE/FALSE` are preferred for boolean values

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**